src/volume.ts

```diff
@@ -138,6 +139,33 @@ export class Volume extends Construct implements IStorage {
     });
   }
 
+  /**
+   * Populate the volume from a SecretProviderClass created as part of a K8s
+   * Secrets Store CSI Driver installation:
+   * https://secrets-store-csi-driver.sigs.k8s.io/introduction.html.
+   *
+   * The Secret Store CSI driver will need an associated provider to source the
+   * secrets. The following provider allows sourcing from AWS Secrets Manager
+   * and Systems Manager Parameter Store:
+   * https://aws.github.io/secrets-store-csi-driver-provider-aws/
+   *
+   * @param secretProvider The secret provider class to use to populate the volume.
+   * @param options Options
+   */
+  public static fromSecretProvider(
+    scope: Construct, id: string, secretProvider: ss.SecretProviderClass, options: SecretProviderVolumeOptions = { }): Volume {
```

Comment:
This will only type match if people use the `SecretProviderClass` from within CDK8s+, which is definitely not ideal. I'm thinking I'll do like the PVC ones and just pass in the name.